addons/default/visiosoft/advs-module/src/Http/Controller/advsController.php

```diff
@@ -151,29 +151,41 @@ public function index($category = null, $city = null)
         $isActiveDopings = $this->adv_model->is_enabled('dopings');
 
         // Search by category slug
+        $categoryId = null;
         if ($category) {
             $categoryId = $this->category_repository->findBy('slug', $category);
-            if ($categoryId) {
-                $param['cat'] = $categoryId->id;
+            if (!$categoryId) {
+                $this->messages->error(trans('visiosoft.module.advs::message.category_not_exist'));
+                return redirect('/');
             }
-        } elseif (isset($param['cat']) && !empty($param['cat'])) {
-            $categoryId = $this->category_repository->find($param['cat']);
-            if ($categoryId) {
+            if (isset($param['cat'])) {
+                unset($param['cat']);
                 return redirect($this->fullLink(
                     $param,
                     route('adv_list_seo', [$categoryId->slug]),
                     array()
                 ));
-            } else {
+            }
+        } elseif (isset($param['cat']) && !empty($param['cat'])) {
+            $categoryId = $this->category_repository->find($param['cat']);
+            if (!$categoryId) {
                 $this->messages->error(trans('visiosoft.module.advs::message.category_not_exist'));
                 return redirect('/');
             }
+            unset($param['cat']);
+            return redirect($this->fullLink(
+                $param,
+                route('adv_list_seo', [$categoryId->slug]),
+                array()
+            ));
         }
 
         // Search by city slug
+        $cityId = null;
         if ($category) {
             if (is_null($city) && isset($param['city'][0]) && !empty($param['city'][0]) && strpos($param['city'][0], ',') === false) {
```

Comment:
Is there anything wrong with the data you are parsing, So many cases ?

Reply:
yes there is something wrong, I am trying to redirect all the possible links to be the same link that shows the category and city slugs instead of showing the params